client/my-sites/checkout/composite-checkout/wechat.js

```diff
@@ -0,0 +1,280 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import styled from '@emotion/styled';
+import debugFactory from 'debug';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@automattic/react-i18n';
+import {
+	Button,
+	usePaymentProcessor,
+	useTransactionStatus,
+	useLineItems,
+	renderDisplayValueMarkdown,
+	useEvents,
+	useFormStatus,
+	registerStore,
+	useSelect,
+	useDispatch,
+} from '@automattic/composite-checkout';
+
+/**
+ * Internal dependencies
+ */
+import { PaymentMethodLogos } from 'client/my-sites/checkout/composite-checkout/wpcom/components/payment-method-logos';
+import Field from 'my-sites/checkout/composite-checkout/wpcom/components/field';
+import {
+	SummaryLine,
+	SummaryDetails,
+} from 'my-sites/checkout/composite-checkout/wpcom/components/summary-details';
+import WeChatPaymentQRcode from 'my-sites/checkout/checkout/wechat-payment-qrcode';
+
+const debug = debugFactory( 'calypso:composite-checkout:wechat-payment-method' );
+
+export function createWeChatPaymentMethodStore() {
+	debug( 'creating a new wechat payment method store' );
+	const actions = {
+		changeCustomerName( payload ) {
+			return { type: 'CUSTOMER_NAME_SET', payload };
+		},
+	};
+
+	const selectors = {
+		getCustomerName( state ) {
+			return state.customerName || '';
+		},
+	};
+
+	const store = registerStore( 'wechat', {
+		reducer(
+			state = {
+				customerName: { value: '', isTouched: false },
+			},
+			action
+		) {
+			switch ( action.type ) {
+				case 'CUSTOMER_NAME_SET':
+					return { ...state, customerName: { value: action.payload, isTouched: true } };
+			}
+			return state;
+		},
+		actions,
+		selectors,
+	} );
+
+	return { ...store, actions, selectors };
+}
+
+export function createWeChatMethod( { store, stripe, stripeConfiguration } ) {
+	return {
+		id: 'wechat',
+		label: <WeChatLabel />,
+		activeContent: <WeChatFields stripe={ stripe } stripeConfiguration={ stripeConfiguration } />,
+		submitButton: (
+			<WeChatPayButton
+				store={ store }
+				stripe={ stripe }
+				stripeConfiguration={ stripeConfiguration }
+			/>
+		),
+		inactiveContent: <WeChatSummary />,
+		getAriaLabel: () => 'WeChat Pay',
+	};
+}
+
+function WeChatFields() {
+	const { __ } = useI18n();
+
+	const customerName = useSelect( ( select ) => select( 'wechat' ).getCustomerName() );
+	const { changeCustomerName } = useDispatch( 'wechat' );
+	const { formStatus } = useFormStatus();
+	const isDisabled = formStatus !== 'ready';
+
+	return (
+		<WeChatFormWrapper>
+			<WeChatField
+				id="cardholderName"
+				type="Text"
+				autoComplete="cc-name"
+				label={ __( 'Your name' ) }
+				value={ customerName?.value ?? '' }
+				onChange={ changeCustomerName }
+				isError={ customerName?.isTouched && customerName?.value.length < 3 }
+				errorMessage={ __( 'Your name must contain at least 3 characters' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Your Name must contain at least 3 characters' )` *ES Score: 9*